src/encoded/static/components/matrix_deeply_profiled.js

```diff
@@ -0,0 +1,576 @@
+            <div className="matrix__presentation">
+                <div className={`matrix__label matrix__label--horz${!scrolledRight ? ' horz-scroll' : ''}`}>
+                    <span>{context.matrix.x.label}</span>
+                    {svgIcon('largeArrow')}
+                </div>
+                <div className="matrix__presentation-content">
+                    <div className="matrix__label matrix__label--vert"><div>{svgIcon('largeArrow')}{context.matrix.y.label}</div></div>
+                    <div className="matrix__data-wrapper">
+                        <div className="matrix__data" onScroll={this.handleOnScroll} ref={(element) => { this.scrollElement = element; }}>
+                            <DataTable tableData={matrixConfig} />
+                        </div>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
+
+MatrixPresentation.propTypes = {
+    /** Matrix search result object */
+    context: PropTypes.object.isRequired,
+    /** Callback to retrieve row categories */
+    rowCategoryGetter: PropTypes.func.isRequired,
+    /** Callback to retrieve subcategories */
+    rowSubCategoryGetter: PropTypes.func.isRequired,
+    /** Callback to map row category query values */
+    mapRowCategoryQueries: PropTypes.func.isRequired,
+    /** Callback to map subcategory query values */
+    mapSubCategoryQueries: PropTypes.func.isRequired,
+};
+
+
+/**
+ * Render the vertical facets and the matrix itself.
+ */
+const MatrixContent = ({ context, rowCategoryGetter, rowSubCategoryGetter, mapRowCategoryQueries, mapSubCategoryQueries }) => (
+    <div className="matrix__content">
+        <MatrixVerticalFacets context={context} />
+        <MatrixPresentation context={context} rowCategoryGetter={rowCategoryGetter} rowSubCategoryGetter={rowSubCategoryGetter} mapRowCategoryQueries={mapRowCategoryQueries} mapSubCategoryQueries={mapSubCategoryQueries} />
+    </div>
+);
+
+MatrixContent.propTypes = {
+    /** Matrix search result object */
+    context: PropTypes.object.isRequired,
+    /** Callback to retrieve row categories from matrix data */
+    rowCategoryGetter: PropTypes.func.isRequired,
+    /** Callback to retrieve subcategories from matrix data */
+    rowSubCategoryGetter: PropTypes.func.isRequired,
+    /** Callback to map row category query values */
+    mapRowCategoryQueries: PropTypes.func.isRequired,
+    /** Callback to map subcategory query values */
+    mapSubCategoryQueries: PropTypes.func.isRequired,
+};
+
+
+/**
+ * Map query values to a query-string component actually used in experiment matrix row category
+ * link queries.
+ * @param {string} rowCategory row category value to map
+ * @param {object} rowCategoryBucket Matrix search result row bucket object
+ *
+ * @return {string} mapped row category query
+ */
+const mapRowCategoryQueriesExperiment = (rowCategory, rowCategoryBucket) => (
+    `${rowCategory}=${encoding.encodedURIComponentOLD(rowCategoryBucket.key)}`
```

Comment:
The comment for encodedURIComponentOLD says, “Do not use this function; all usage should move to encodedURIComponent.”